From c0e4cf4aa95f175c83249b1e017cab75945cd0ba Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:23:03 -0400
Subject: [x86] pci: Use dev_is_pci() to identify PCI devices

Message-id: <20140719042303.19438.47337.stgit@gir.stowe>
Patchwork-id: 85965
O-Subject: [RHEL7.1 PATCH 03/81] x86/PCI: Use dev_is_pci() to identify PCI devices
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 894d33437872d974c636da55563a8474fa6713c5

commit 894d33437872d974c636da55563a8474fa6713c5
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Dec 5 19:54:34 2013 +0800

    x86/PCI: Use dev_is_pci() to identify PCI devices

    Use dev_is_pci() instead of checking bus type directly.

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/acpi/boot.c b/arch/x86/kernel/acpi/boot.c
index fead832..45a51de 100644
--- a/arch/x86/kernel/acpi/boot.c
+++ b/arch/x86/kernel/acpi/boot.c
@@ -1091,9 +1091,7 @@ static int mp_config_acpi_gsi(struct device *dev, u32 gsi, int trigger,
 
  if (!acpi_ioapic)
   return 0;
- if (!dev)
-  return 0;
- if (dev->bus != &pci_bus_type)
+ if (!dev || !dev_is_pci(dev))
   return 0;
 
  pdev = to_pci_dev(dev);
-- 
1.7.1